app/components/projects/table_component.rb

```diff
@@ -195,6 +195,14 @@ def favored_project_ids
       @favored_project_ids ||= Favorite.where(user: current_user, favored_type: "Project").pluck(:favored_id)
     end
 
+    def project_life_cycle_step_by_definition(definition, project)
+      @project_life_cycle_steps_by_definition ||= Project::LifeCycleStep
+                                                    .where(active: true)
+                                                    .group_by { |s| [s.definition_id, s.project_id] }
+
+      @project_life_cycle_steps_by_definition[[definition.id, project.id]]&.first
```

Comment:
Sorry, I forgot that we have only one LifeCycleStep per Project and per Definition. In this case, we can use `index_by` instead of `group_by`. `index_by` just takes the latest value for the key and assigns that, instead of aggregating them into an array. That way the accessing looks a bit less ugly.

```suggestion
                                                    .index_by { |s| [s.definition_id, s.project_id] }

      @project_life_cycle_steps_by_definition[[definition.id, project.id]]
```